test/rpc/rpc-responseSpec.ts

```diff
@@ -0,0 +1,140 @@
+import { expect } from 'chai'
+import * as sinon from 'sinon'
+import { getServicesMock } from '../mocks'
+import { EVENT } from '../../src/constants'
+import { TOPIC, RPC_ACTIONS, RPCMessage } from '../../binary-protocol/src/message-constants'
+
+import { DefaultOptions } from '../../src/client-options'
+import { RPCResponse } from '../../src/rpc/rpc-response'
+
+describe('RPC response', () => {
+  let services: any
+  let rpcResponse: RPCResponse
+  const name = 'myRPC'
+  const correlationId = 'correlationId'
+
+  beforeEach(() => {
+    services = getServicesMock()
+    rpcResponse = new RPCResponse({
+        topic: TOPIC.RPC,
+        action: RPC_ACTIONS.REQUEST,
+        name,
+        correlationId
+      },
+      DefaultOptions,
+      services
+    )
+    rpcResponse.autoAccept = false
+  })
+
+  afterEach(() => {
+    services.connectionMock.verify()
+  })
+
+  it('doesn\'t accept automatically when autoAccept == false', done => {
+    services.connectionMock
+      .expects('sendMessage')
+      .never()
+
+    process.nextTick(done)
+  })
+
+  it('sends an accept message automatically when autoAccept == true ', done => {
+    services.connectionMock
+      .expects('sendMessage')
+      .once()
+      .withExactArgs({
+        topic: TOPIC.RPC,
+        action: RPC_ACTIONS.ACCEPT,
+        name,
+        correlationId
+      })
+
+    rpcResponse.autoAccept = true
+    process.nextTick(done)
+  })
+
+  it('sends an accept message manually', () => {
+    services.connectionMock
+      .expects('sendMessage')
+      .once()
+      .withExactArgs({
+        topic: TOPIC.RPC,
+        action: RPC_ACTIONS.ACCEPT,
+        name,
+        correlationId
+      })
+
+    rpcResponse.accept()
+  })
+
+  it('sends the response message', () => {
+    const data = { foo: 'bar'}
+    const message = {
+      topic: TOPIC.RPC,
+      action: RPC_ACTIONS.RESPONSE,
+      name,
+      correlationId,
+      parsedData: data
+    }
+    services.connectionMock
+      .expects('sendMessage')
+      .once()
+    services.connectionMock
+      .expects('sendMessage')
+      .once()
+      .withExactArgs(message)
+
+    rpcResponse.send(data)
+  })
+
+  it('throws when trying to send a completed response', () => {
+    const data = { foo: 'bar'}
+    services.connectionMock
+      .expects('sendMessage')
+      .twice()
+
+    rpcResponse.send(data)
+    expect(rpcResponse.send.bind(rpcResponse, data)).to.throw()
+  })
+
+  it('doesn\'t send multiple accept messages', () => {
+    services.connectionMock
+      .expects('sendMessage')
+      .once()
+
+    rpcResponse.accept()
+    rpcResponse.accept()
+  })
+
+  it('sends reject message', () => {
+    services.connectionMock
```

Comment:
rejecting twice

Reply:
Did that just work with once and twice? If so something is wrong with test verification.

Reply:
Sorry Yasser, but in which test do you mean once and twice now?

Reply:
The test initially had .twice and now its .once

Why did it work with .twice before?